src/prommis/nanofiltration/costing/diafiltration.py

```diff
@@ -0,0 +1,713 @@
+#########################################################
+# MSContactor flowsheet code adapted from Andrew Lee.
+#########################################################
+
+
+import pyomo.environ as pyo
+from pyomo.network import Arc
+
+from idaes.core.util.model_diagnostics import DiagnosticsToolbox
+
+from idaes.core import (
+    Component,
+    declare_process_block_class,
+    FlowsheetBlock,
+    MaterialBalanceType,
+    MaterialFlowBasis,
+    Phase,
+    PhysicalParameterBlock,
+    StateBlock,
+    StateBlockData,
+    UnitModelBlock,
+    UnitModelCostingBlock,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, propagate_state
+from prommis.nanofiltration.costing.diafiltration_cost_model import (
+    DiafiltrationCosting,
+    DiafiltrationCostingData,
+)
+
+from idaes.models.unit_models import (
```

Comment:
You should group imports by type, and keep them in the following order:

1. Native Python imports
2. Third part Python imports
3. Pyomo imports
4. IDAES imports
5. PrOMMiS imports.

It can also be helpful to keep each section in alphabetical order, if only to make it easier to find each import.